src/hooks/cgroups/pbs_cgroups.PY

```diff
@@ -1098,11 +1100,28 @@ class JobUtils(object):
                             str(chunk.chunk_resources[resc])
         if resources:
             pbs.logmsg(pbs.EVENT_DEBUG4, '%s: Resources for %s: %s' %
-                       (caller_name(), self.hostname, repr(resources)))
-        else:
-            pbs.logmsg(pbs.EVENT_DEBUG2,
-                       '%s: No resources assigned to host %s' %
-                       (caller_name(), self.hostname))
+                       (caller_name(), hostname, repr(resources)))
+            # Return assigned resources for specified host
+            return resources
+        # Workaround for systems where node is short hostname
+        pbs.logmsg(pbs.EVENT_DEBUG2,
+                   '%s: No resources assigned to host %s' %
+                   (caller_name(), hostname))
+        try:
+            cmd = ['hostname', '-s']
+            process = subprocess.Popen(cmd, shell=False,
+                                       stdout=subprocess.PIPE,
+                                       stderr=subprocess.PIPE)
+            out, err = process.communicate()
+        except KeyboardInterrupt:
+            raise
+        except:
```

Comment:
According to python's docs, Popen() will raise OSError, ValueError, or CalledProcessError.  Consider not doing a blanket except and only looking for those.